chapter_5/find_module/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+# CMakeLists to show how find-modules work
+#
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.21)
+
+project(
+    ch5_find_module_example
+    VERSION 1.0
+    DESCRIPTION
+        "A simple C++ project to demonstrate how find-modules work"
+    LANGUAGES CXX
+)
+
+# the following code is used to set up the library which we want to find with the  find_module.
+# calling cmake from itself is dangerous and should not be done in practice
+# this will configure and build the hello_lib from chapter3 into the folder ./dep
+# it will not perform a regular installation as the idea is to provide an example of how to find non-cmake projects
+
+execute_process(
```

Comment:
I think these three steps can be avoided by using `ExternalProject_Add` instead. And about the "idea is to provide an example of how to find non-cmake projects" part, I am a bit confused. The `obscure` is still a CMake project. Maybe the wording can be changed to external CMake projects, or am I missing something?

Reply:
The comment is misleading, I think I have to change it. Maybe I also m8ve this block to somewhere else.

The goal of this is to find a binary dependency that is not in a standard location using a custom find module. ExternalProject does not work here, because with it the dowbload an building happens only at build time, but for the find module it has to be there at configuration time. FetchContent does also not work for similar reasons. 

It is a bit of a hack how I solved it, but I did not find another way to ensure that there is compatible binary available in a non-standard location across all systems. Unless we precompile and check in the binaries into git, which I think is prone to error. 
If you know a better way I'm open to it. I find this way confusing and ugly as well.